librosa/core/audio.py

```diff
@@ -521,6 +524,127 @@ def zero_crossings(y, threshold=1e-10, ref_magnitude=None, pad=True,
+    """Returns a signal with the signal `click` placed at each specified time
+
+    Parameters
+    ----------
+    times : np.ndarray or None
+        times to place clicks, in seconds
+
+    frames : np.ndarray or None
+        frame indices to place clicks
+
+    sr : number
+        desired sampling rate of the output signal
+
+    hop_length : int > 0
+        if positions are specified by `frames`, the number of samples between frames.
+
+    click_freq : float > 0
+        frequency (in Hz) of the default click signal.  Default is 1KHz.
+
+    click_duration : float > 0
+        duration (in seconds) of the default click signal.  Default is 100ms.
+
+    click : np.ndarray or None
+        optional click signal sample to use instead of the default blip.
+
+    length : int > 0
+        desired number of samples in the output signal,
+
+
+    Returns
+    -------
+    click_signal : np.ndarray
+        Synthesized click signal
+
+
+    Raises
+    ------
+    ParameterError
+        - If neither `times` nor `frames` are provided.
+        - If any of `click_freq`, `click_duration`, or `length` are out of range.
+
+
+    Examples
+    --------
+    >>> # Sonify detected beat events
+    >>> y, sr = librosa.load(librosa.util.example_audio_file())
+    >>> tempo, beats = librosa.beat.beat_track(y=y, sr=sr)
+    >>> y_beats = librosa.clicks(frames=beats, sr=sr)
+
+    >>> # Or generate a signal of the same length as y
+    >>> y_beats = librosa.clicks(frames=beats, sr=sr, length=len(y))
+
+    >>> # Or use timing instead of frame indices
+    >>> times = librosa.frames_to_time(beats, sr=sr)
+    >>> y_beat_times = librosa.clicks(times=times, sr=sr)
+
+    >>> # Or with a click frequency of 880Hz and a 500ms sample
+    >>> y_beat_times880 = librosa.clicks(times=times, sr=sr,
+    ...                                  click_freq=880, click_duration=0.5)
+    """
+
+    # Compute sample positions from time or frames
+    if times is None:
+        if frames is None:
+            raise ParameterError('either "times" or "frames" must be provided')
+
+        positions = frames_to_samples(frames, hop_length=hop_length)
+    else:
+        # Convert times to positions
+        positions = time_to_samples(times, sr=sr)
+
+    if click is not None:
+        # Check that we have a well-formed audio buffer
+        util.valid_audio(click, mono=True)
+
+    else:
+        # Create default click signal
+        if click_duration <= 0:
+            raise ParameterError('click_duration must be strictly positive')
+
+        if click_freq <= 0:
+            raise ParameterError('click_freq must be strictly positive')
+
+        angular_freq = 2 * np.pi * click_freq / float(sr)
+
+        click = np.logspace(0, -10,
+                            num=int(np.round(sr * click_duration)),
+                            base=2.0)
+
+        click *= np.sin(angular_freq * np.arange(len(click)))
```

Comment:
Yeah, fair enough.  It's not obvious to me yet that it's worth factoring out, but I'll consider it.